From: Pietro Ameruoso <p.ameruoso@live.it>
Date: Mon, 17 Aug 2026 00:00:00 +0200
Subject: [PATCH] arm64: dts: mediatek: mt7988a-bpi-r4-pro: use PHY LED framework

Describe the LEDs of both external Aeonsemi PHYs using the common PHY
LED binding. Assign color and function properties so phylib can create
unique LED class device names and offload the netdev trigger to the PHY.

Signed-off-by: Pietro Ameruoso <p.ameruoso@live.it>
---
--- a/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4-pro-8x.dts
+++ b/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4-pro-8x.dts
@@ -30,18 +30,20 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 
-			phy24_led0: led0@0 {
+			led@0 {
 				reg = <0>;
 				linux,default-trigger = "netdev";
-				active-high;
-				status = "okay";
+				color = <LED_COLOR_ID_GREEN>;
+				function = LED_FUNCTION_LAN;
+				active-low;
 			};
 
-			phy24_led1: led1@1 {
+			led@1 {
 				reg = <1>;
 				linux,default-trigger = "netdev";
+				color = <LED_COLOR_ID_YELLOW>;
+				function = LED_FUNCTION_LAN;
 				active-high;
-				status = "okay";
 			};
 		};
 	};
@@ -59,18 +61,20 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 
-			phy28_led0: led0@0 {
+			led@0 {
 				reg = <0>;
 				linux,default-trigger = "netdev";
-				active-high;
-				status = "okay";
+				color = <LED_COLOR_ID_GREEN>;
+				function = LED_FUNCTION_WAN;
+				active-low;
 			};
 
-			phy28_led1: led1@1 {
+			led@1 {
 				reg = <1>;
 				linux,default-trigger = "netdev";
+				color = <LED_COLOR_ID_YELLOW>;
+				function = LED_FUNCTION_WAN;
 				active-high;
-				status = "okay";
 			};
 		};
 	};
